packages/wonder-blocks-dropdown/components/multi-select.test.js

```diff
@@ -281,4 +285,194 @@ describe("MultiSelect", () => {
+        select.setState({open: true});
+        expect(
+            select
+                .find(ActionItem)
+                .at(0)
+                .exists(),
+        ).toBe(true);
+
+        // Assert
+        select.setState({searchText: "2"});
+
+        expect(
+            select
+                .find(ActionItem)
+                .at(0)
+                .exists(),
+        ).toBe(false);
+    });
+
+    it("arrow up/down moves focus from the search text input like other items", () => {
+        // Arrange
+        select.setProps({isFilterable: true});
+        select.setState({open: true});
+        const searchInput = select.find(SearchTextInput);
+
+        const selectAll = select
+            .find(ActionItem)
+            .at(0)
+            .find(ClickableBehavior);
+        const selectNone = select
+            .find(ActionItem)
+            .at(1)
+            .find(ClickableBehavior);
+
+        const item1 = select
+            .find(OptionItem)
+            .at(0)
+            .find(ClickableBehavior);
+        const item3 = select
+            .find(OptionItem)
+            .at(2)
+            .find(ClickableBehavior);
+
+        // Act and Assert
+        // The focus is on opener. Press up (or down) should focus the input
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(searchInput.state("focused")).toBe(true);
+
+        // one more up should move focus to the last item
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(item3.state("focused")).toBe(true);
+
+        // two more up should move focus twice
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(item1.state("focused")).toBe(true);
+
+        // one more up should move focus to the select none
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(selectNone.state("focused")).toBe(true);
+
+        // one more up should move focus to the select all
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(selectAll.state("focused")).toBe(true);
+
+        // one more up should move focus to the input again.
+        select.simulate("keydown", {keyCode: keyCodes.up});
+        select.simulate("keyup", {keyCode: keyCodes.up});
+        expect(searchInput.state("focused")).toBe(true);
+
+        // Press keyboard down arrow should move the focus down
+        select.simulate("keydown", {keyCode: keyCodes.down});
+        select.simulate("keyup", {keyCode: keyCodes.down});
+        expect(selectAll.state("focused")).toBe(true);
```

Comment:
Multiple expects and acts in the same test is really hard to maintain.

I think that these should be separate test cases to keep things simple and clear. As it stands, if one of the early expectations fails, we never test the remainder, which can make tracking down why something fails much harder.

What do you think about splitting this into multiple, specific cases either as separate `it` calls or a combination of `it.each` uses?